epictrack-api/src/api/services/user.py

```diff
@@ -21,45 +21,73 @@
 class UserService:
     """User Service"""
 
-    @staticmethod
-    def get_all_users():
+    @classmethod
+    def get_all_users(cls):
         """Get all users"""
         users = KeycloakService.get_users()
         for user in users:
-            user['group'] = None
+            user["group"] = None
         groups = UserService.get_groups()
-        groups = sorted(groups, key=lambda x: x['attributes']['level'][0])
+        groups = sorted(groups, key=UserService._get_level)
         for group in groups:
-            members = KeycloakService.get_group_members(group['id'])
-            member_ids = [member['id'] for member in members]
-            filtered_users = list(filter(lambda x, _member_ids=member_ids: x['id'] in _member_ids, users))
+            members = KeycloakService.get_group_members(group["id"])
+            member_ids = [member["id"] for member in members]
+            filtered_users = [user for user in users if user["id"] in member_ids]
             for user in filtered_users:
-                user['group'] = group
+                user["group"] = group
         return users
 
-    @staticmethod
-    def get_groups():
+    @classmethod
+    def get_groups(cls):
         """Get groups that has level set up"""
         groups = KeycloakService.get_groups()
-        filtered_groups = list(filter((lambda g: 'level' in g['attributes']), groups))
+        filtered_groups = []
+        for group in groups:

```

Comment:
can we simplify the nesting, and perhaps break this into different functions that makes it easier to understand. Suggestion:

    @classmethod
    def get_groups_with_level(cls):
        """Get groups that have a 'level' attribute set up."""
        all_groups = KeycloakService.get_groups()
        sub_groups_with_level = cls._get_sub_groups_with_level(all_groups)
        groups_with_level = cls._get_groups_with_level(all_groups)

        return sub_groups_with_level + groups_with_level

    @classmethod
    def _get_sub_groups_with_level(cls, all_groups):
        """Get sub-groups with 'level' attribute."""
        return [
            sub_group
            for group in all_groups
            for sub_group in group.get("subGroups", [])
            if "level" in sub_group.get("attributes", {})
        ]

    @classmethod
    def _get_groups_with_level(cls, all_groups):
        """Get groups with 'level' attribute."""
        return [
            group
            for group in all_groups
            if "level" in group.get("attributes", {})
        ]